packages/cart/integration-tests/__tests__/services/cart-module/index.spec.ts

```diff
@@ -1860,4 +1824,476 @@ describe("Cart Module Service", () => {
       expect(adjustments?.length).toBe(0)
     })
   })
+
+  describe("setLineItemTaxLines", () => {
+    it("should set line item tax lines for a cart", async () => {
+      const [createdCart] = await service.create([
+        {
+          currency_code: "eur",
+        },
+      ])
+
+      const [itemOne] = await service.addLineItems(createdCart.id, [
+        {
+          quantity: 1,
+          unit_price: 100,
+          title: "test",
+        },
+      ])
+
+      const [itemTwo] = await service.addLineItems(createdCart.id, [
+        {
+          quantity: 2,
+          unit_price: 200,
+          title: "test-2",
+        },
+      ])
+
+      const taxLines = await service.setLineItemTaxLines(createdCart.id, [
+        {
+          item_id: itemOne.id,
+          rate: 20,
+          code: "TX",
+        },
+        {
+          item_id: itemTwo.id,
+          rate: 20,
+          code: "TX",
+        },
+      ])
+
+      expect(taxLines).toEqual(
+        expect.arrayContaining([
+          expect.objectContaining({
+            item_id: itemOne.id,
+            rate: 20,
+            code: "TX",
+          }),
+          expect.objectContaining({
+            item_id: itemTwo.id,
+            rate: 20,
+            code: "TX",
+          }),
+        ])
+      )
+    })
+
+    it("should replace line item tax lines for a cart", async () => {
+      const [createdCart] = await service.create([
+        {
+          currency_code: "eur",
+        },
+      ])
+
+      const [itemOne] = await service.addLineItems(createdCart.id, [
+        {
+          quantity: 1,
+          unit_price: 100,
+          title: "test",
+        },
+      ])
+
+      const taxLines = await service.setLineItemTaxLines(createdCart.id, [
+        {
+          item_id: itemOne.id,
+          rate: 20,
+          code: "TX",
+        },
+      ])
+
+      expect(taxLines).toEqual(
+        expect.arrayContaining([
+          expect.objectContaining({
+            item_id: itemOne.id,
+            rate: 20,
+            code: "TX",
+          }),
+        ])
+      )
+
+      await service.setLineItemTaxLines(createdCart.id, [
+        {
+          item_id: itemOne.id,
+          rate: 25,
+          code: "TX-2",
+        },
+      ])
+
+      const cart = await service.retrieve(createdCart.id, {
+        relations: ["items.tax_lines"],
+      })
+
+      expect(cart.items).toEqual(
+        expect.arrayContaining([
+          expect.objectContaining({
+            id: itemOne.id,
+            tax_lines: expect.arrayContaining([
+              expect.objectContaining({
+                item_id: itemOne.id,
+                rate: 25,
+                code: "TX-2",
+              }),
+            ]),
+          }),
+        ])
+      )
+
+      expect(cart.items?.length).toBe(1)
+      expect(cart.items?.[0].tax_lines?.length).toBe(1)
+    })
+
+    it("should remove all line item tax lines for a cart", async () => {
```

Comment:
thought(non-blocking): maybe add a case that would trigger both a delete, update and create.